src/Common/src/System/Net/SafeCloseSocket.Windows.cs

```diff
@@ -39,44 +33,37 @@ public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipComp
                 throw new ObjectDisposedException(typeof(Socket).FullName);
             }
 
-            // Check to see if the socket native _handle is already
-            // bound to the ThreadPool's completion port.
-            if (_iocpBoundHandle == null)
+            // Lazily bind the socket native _handle to the ThreadPool's completion port.
+            LazyInitializer.EnsureInitialized(ref _iocpBoundHandle, ref _iocpBindingLock, () =>
```

Comment:
Undo this.  This is adding a delegate allocation and closure allocation on every call.